test/python/custom_op_shape_dtype_fn.py

```diff
@@ -0,0 +1,71 @@
+import os
+import tempfile
+
+import torch
+import torch.utils.cpp_extension
+import torch_mlir
+from torch_mlir_e2e_test.annotations import export, annotate_args
+
+
+# RUN: %PYTHON %s | FileCheck %s
+
+
+def identity(_5: torch.Tensor):
+    return _5
+
+
+goofy_lib = torch.library.Library("goofy", "DEF")
+goofy_lib.define("identity(Tensor t) -> Tensor")
+goofy_lib.impl("identity", identity)
+
+
+class CustomOpExampleModule(torch.nn.Module):
+    def __init__(self):
+        super().__init__()
+
+    @export
+    @annotate_args(
+        [
+            None,
+            ([-1, -1], torch.float32, True),
+        ]
+    )
+    def forward(self, a):
+        b = 2 * a
+        return torch.ops.goofy.identity(b)
+
+
+mod = CustomOpExampleModule()
+mod.eval()
+
+abstract_interp_src = """\
+func.func @__torch_mlir_shape_fn.goofy.identity(%arg0: !torch.list<int>) -> !torch.list<int> {
+  return %arg0 : !torch.list<int>
+}
+func.func @__torch_mlir_dtype_fn.goofy.identity(%arg0: !torch.int, %arg1: !torch.int) -> !torch.int {
+  return %arg1 : !torch.int
+}
+func.func @__torch_mlir_has_value_semantics.goofy.identity() { return }
```

Comment:
you need either `{ return }` or `func.func private`. whichever you want is fine with me.